internal/component/artifactory_release_source.go

```diff
@@ -247,15 +247,38 @@ func (ars *ArtifactoryReleaseSource) FindReleaseVersion(spec cargo.BOSHReleaseTa
 		if releases.Folder {
 			continue
 		}
-		versions := semverPattern.FindAllString(filepath.Base(releases.URI), -1)
-		version := versions[0]
-		stemcellVersion := versions[len(versions)-1]
-		version = strings.Replace(version, "-", "", -1)
-		version = strings.Replace(version, "v", "", -1)
-		stemcellVersion = strings.Replace(stemcellVersion, "-", "", -1)
+
+		artifactoryFilename := filepath.Base(releases.URI)
+		versions := semverPattern.FindAllString(artifactoryFilename, -1)
+
+		versionFormatter := strings.NewReplacer("-", "", "v", "")
+
+		replacements := make([]string, 0)
+		for _, ver := range versions {
+			ver = versionFormatter.Replace(ver)
+			replacements = append(replacements, ver)
+			replacements = append(replacements, "")
+		}
+		removeVersions := strings.NewReplacer(replacements...)
+		remotePathFilename := filepath.Base(remotePath)
+
+		// we want to ensure that the path template and the artifactory file being considered
+		// having matching formats. we strip the versions out because those aren't material
+		// to this check
+		if removeVersions.Replace(remotePathFilename) != removeVersions.Replace(artifactoryFilename) {
+			continue
+		}
+
+		stemcellVersion := versionFormatter.Replace(versions[len(versions)-1])
```

Comment:
This appears to make an assumption that the stemcell version is the last version in the pattern. Is that accurate?

Reply:
I believe it is, but it could be made clearer by only using it where it's valid. 

```
if len(versions) > 1 {
	stemcellVersion := versionFormatter.Replace(versions[len(versions)-1])
	if stemcellVersion != spec.StemcellVersion {
		continue
	}
}
```

reusing the existing convention of the last slice entry